apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/remote/GRPCChannelManager.java

```diff
@@ -88,24 +88,22 @@ public void run() {
                 String server = "";
                 try {
                     int index = Math.abs(random.nextInt()) % grpcServers.size();
-                    if (index != selectedIdx) {
-                        selectedIdx = index;
+                    selectedIdx = index;
```

Comment:
Why removed this if?

Reply:
Most of the components (such as `ServiceAndEndpointRegisterClient`) depend on the status change callback `org.apache.skywalking.apm.agent.core.remote.GRPCChannelListener#statusChanged` (triggered by `notify` line 106), if there is only one OAP node and it's restarting, the status would be `DISCONNECTED` and never change back to `CONNECTED`, all the components depends on the status won't work since then

Reply:
OK. That is a bug. The reason was trying to this is, the gRPC will reconnect directly, so if the address has only one, should wait for gRPC to reconnect. We may need to change this code better.

Reply:
Thanks for clarifying this, I try another way to avoid recreating a new channel every time (take the advantage of GRPC's auto-reconnectivity) when there is only one server, by checking the existed channel's state.

Reply:
Fixed in #3181 and I'll mark it as resolved